utilities/deep_inspection.py

```
@@ -96,6 +107,107 @@ def create_di_connection_secret(
+        for sample in TimeoutSampler(
+            wait_timeout=timeout,
+            sleep=3,
+            func=lambda: conversion.instance.status,
+        ):
+            if not sample:
+                continue
+
+            current_phase = sample.get("phase", "")
+            last_phase = current_phase
+            if current_phase == phase:
+                LOGGER.info(f"Conversion '{conversion.name}' reached phase '{phase}'")
```

Ruff (0.15.20)

[warning] Logging statement uses f-string

(G004)